### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by mzanella**
- remove unused variables
 - remove useless comments
 - add comment to .h files
 - reformat .cpp files
 - solved most of the errors/warnings

### CODE REVIEW

File: src/classifier/classifier.cpp [hunk 1]
```diff
@@ -59,15 +55,17 @@ classifier::pkt_type classifier::Classifier::Mapper::default_mapping() {
 
 // ************************** temporary testing stuff **************************
 //TODO to remove when harbour calls are implemented
-classifier::Classifier::Mapper classifier::Classifier::Mapper::defaultUDPMapper() {
-    Mapper m;
+classifier::Classifier::Mapper
```
Comment (Polpetta) [inline src/classifier/classifier.cpp:58]: This is part of a TODO. Do we still need this stuff?

File: src/classifier/classifier.cpp [hunk 2]
```diff
@@ -59,15 +55,17 @@ classifier::pkt_type classifier::Classifier::Mapper::default_mapping() {
...
+classifier::Classifier::Mapper::defaultUDPMapper() {
+    Mapper m{};
     m.type_ = pkt_type::UDP;
     return m;
 }
 
 //TODO to remove when harbour calls are implemented
-classifier::Classifier::Mapper classifier::Classifier::Mapper::defaultTCPMapper() {
-    Mapper m;
+classifier::Classifier::Mapper
```
Comment (Polpetta) [inline src/classifier/classifier.cpp:66]: This is part of a TODO. Do we still need this stuff?

File: src/db/dbquery.cpp [hunk 3]
```diff
@@ -58,7 +54,8 @@ std::string db::DBQuery::create_entry(const Query& query) {
         curl_easy_setopt(curl, CURLOPT_URL, req_addr.c_str());
         curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "PUT");
         //curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L);
```
Comment (Polpetta) [inline src/db/dbquery.cpp]: This should be cleaned too

```suggestion

```

File: src/db/dbquery.cpp [hunk 4]
```diff
@@ -58,7 +54,8 @@ std::string db::DBQuery::create_entry(const Query& query) {
...
-        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, db::DBQuery::curl_callback);
+        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION,
+                         db::DBQuery::curl_callback);
         curl_easy_setopt(curl, CURLOPT_WRITEDATA, &req_data_res);
         //curl_easy_setopt(curl, CURLOPT_USERAGENT, "libcurl-agent/1.0");
```
Comment (Polpetta) [inline src/db/dbquery.cpp]: This should be cleaned too
```suggestion

```

File: src/db/dbquery.cpp [hunk 5]
```diff
@@ -111,22 +108,24 @@ bool db::DBQuery::update_endpoint(const Query& query, const Endpoint& to_add) {
         curl_easy_setopt(curl, CURLOPT_URL, req_addr.c_str());
         curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "PUT");
         //curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L);
```
Comment (Polpetta) [inline src/db/dbquery.cpp]: This should be cleaned too
```suggestion

```

File: src/db/dbquery.cpp [hunk 6]
```diff
@@ -111,22 +108,24 @@ bool db::DBQuery::update_endpoint(const Query& query, const Endpoint& to_add) {
...
-        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, db::DBQuery::curl_callback);
+        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION,
+                         db::DBQuery::curl_callback);
         curl_easy_setopt(curl, CURLOPT_WRITEDATA, &req_data_res);
         //curl_easy_setopt(curl, CURLOPT_USERAGENT, "libcurl-agent/1.0");
```
Comment (Polpetta) [inline src/db/dbquery.cpp]: This should be cleaned too
```suggestion

```

File: src/db/dbquery.cpp [hunk 7]
```diff
@@ -251,9 +255,9 @@ std::vector<db::utils::Address> db::DBQuery::get_route_list(uint32_t p_id) {
             /*for (Json::Value::ArrayIndex i = 0; i != response.size(); i++) {
                 LOG(ltrace, response[i]["port"].asString());
             }*/
```
Comment (Polpetta) [inline src/db/dbquery.cpp]: This piece of commented code should be deleted

File: src/db/dbquery.h [hunk 8]
```diff
@@ -58,27 +54,68 @@ const char* const ROUTE_PREFIX = "routes/";
...
 
+/**
+ * Class to perform queries to roulette
+ */
 class DBQuery {
 private:
     const utils::Address roulette_addr;
 
+    /**
+     * Checks if operation is performed successfully
```
Comment (Polpetta) [inline src/db/dbquery.h]: English
```suggestion
     * Checks if the operation is performed successfully
```

File: src/db/dbquery.h [hunk 9]
```diff
@@ -58,27 +54,68 @@ const char* const ROUTE_PREFIX = "routes/";
...
+    /**
+     * cleanup CURL resources
+     */
     static void clear_local_res(CURL*);
 public:
-    DBQuery(const std::string&, uint16_t port);
-    DBQuery(const utils::Address&);
+    /**
+     * Constructor
+     * @param ip String that represent the IP of roulette
```
Comment (Polpetta) [inline src/db/dbquery.h]:
```suggestion
     * @param ip String that represents the Roulette IP
```

File: src/db/dbquery.h [hunk 10]
```diff
@@ -58,27 +54,68 @@ const char* const ROUTE_PREFIX = "routes/";
...
+     * @param port Port used by roulette
+     */
+    DBQuery(const std::string& ip, uint16_t port);
+    /**
+     * Constructor
+     * @param address of roulette
```
Comment (Polpetta) [inline src/db/dbquery.h]:
```suggestion
     * @param Roulette address
```

File: src/db/dbquery.h [hunk 11]
```diff
@@ -166,11 +202,30 @@ class DBQuery {
 
     typedef Query Entry;
 
+    /**
+     * Create entry on roulette for endpoints
```
Comment (Polpetta) [inline src/db/dbquery.h]:
```suggestion
     * Adds an endpoint entry on Roulette
```

File: src/db/dbquery.h [hunk 12]
```diff
@@ -166,11 +202,30 @@ class DBQuery {
...
+     * @param query to add an entry to endpoints
```
Comment (Polpetta) [inline src/db/dbquery.h]:
```suggestion
     * @param query to add an endpoint entry
```

File: src/db/utils/address.cpp [hunk 13]
```diff
@@ -1,39 +1,36 @@
...
 }
 
-void db::utils::Address::address_parse(const std::string& url) {
+void db::utils::Address::address_parse(const std::string &url) {
     size_t found = url.find_first_of(PORT_SEPARATOR);
-    address = url.substr(0,found);
+    address = url.substr(0, found);
     size_t found1 = url.find_first_of(PATH_SEPARATOR);
-    port = std::stoul(url.substr(found+1,found1-found-1));
+    port = (uint16_t) (std::stoul(url.substr(found + 1, found1 - found - 1)));
```
Comment (Polpetta) [inline src/db/utils/address.cpp]: Should use `static_const` but whatever, our code is full of C casts (so it's really not necessary)...
```suggestion
    port = static_cast<uint16_t>(std::stoul(url.substr(found + 1, found1 - found - 1)));
```

File: src/db/dbquery.cpp [hunk 14]
```diff
@@ -407,39 +414,39 @@ std::string db::DBQuery::Query::to_json() const {
...
     json_res[query::endpoint::DST_PORT] = port_dst;
     json_res[query::endpoint::SFC_ID] = id_sfc;
-    (prt == protocol_type::TCP)? json_res[query::endpoint::PROTOCOL] = "tcp" :
+    (prt == protocol_type::TCP) ? json_res[query::endpoint::PROTOCOL] = "tcp" :
             json_res[query::endpoint::PROTOCOL] = "udp";
-    for (auto it = endpoints.cbegin(); it != endpoints.cend(); it++) {
-        if (it->get_endpoint_typology() == INGRESS_T) {
-            json_res[query::endpoint::SOCK_INGRESS] = it->get_socket_id();
-            json_res[query::endpoint::INGRESS_IP] = it->get_ip();
+    for (auto e : endpoints) {
```
Comment (Polpetta) [inline src/db/dbquery.cpp]: `e` could be `const`:
```suggestion
    for (const auto e : endpoints) {
```

File: src/endpoint/egress.cpp [hunk 15]
```diff
@@ -78,40 +54,34 @@ void endpoint::Egress::manage_exiting_udp_packets(unsigned char* pkt,
...
+        LOG(ldebug,
+            "Packet Send "/* + std::to_string(getsockname(raw_socket, (sockaddr*)&sockstr, &socklen))*/);
     }
 
-    LOG(ldebug, std::to_string(getsockname(raw_socket, (sockaddr*)&sockstr, &socklen)));
+    LOG(ldebug, std::to_string(
+            getsockname(raw_socket, (sockaddr *) &sockstr, &socklen)));
 
     /*socket = client.send_only(pkt + pkt_calc, // move pointer after headers
                               pkt_len - pkt_calc, // resize without considering headers
```
Comment (Polpetta) [inline src/endpoint/egress.cpp]: This commented code should be removed

File: src/endpoint/egress.cpp [hunk 16]
```diff
@@ -324,126 +251,36 @@ void endpoint::Egress::manage_exiting_tcp_packets(unsigned char* pkt,
...
-
-    }  while(1);*/
-
-    //free(pkt);
-    //free(buffer);
+    } while (true);
 
     LOG(ldebug, "exit managing TCP exiting pkts");
 
     // TODO think when closing socket
```
Comment (Polpetta) [inline src/endpoint/egress.cpp:249]: It's this TODO done?

File: src/endpoint/egress.h [hunk 17]
```diff
@@ -24,17 +26,49 @@
 
 namespace endpoint {
 
-
-
 class Egress : public Endpoint {
 private:
+    /**
+     * Method used to manage UDP packets received from the chain.
```
Comment (Polpetta) [inline src/endpoint/egress.h]: _Used to_ [refers to hold habits in the past](https://dictionary.cambridge.org/it/grammatica/grammatica-britannico/past/used-to), it should not be used here
```suggestion
     * It manages incoming UDP packets that it has received from the chain
```

File: src/endpoint/egress.h [hunk 18]
```diff
@@ -24,17 +26,49 @@
...
+     * Once a packet is received the packet is read (SFC header and IP header)
+     * in order to manage it differently depending on the protocol used.
+     * @param args udp_pkt_mngmnt_args struct that contains data to manage
+     * packets from the chain.
+     */
     void manage_pkt_from_chain(void* args);
+    /**
+     * Method used to manage TCP connection with the destination.
```
Comment (Polpetta) [inline src/endpoint/egress.h]: Same
```suggestion
     * The method handles a TCP connection with a defined destination
```

File: src/endpoint/egress.h [hunk 19]
```diff
@@ -24,17 +26,49 @@
...
+     * First the method checks if there is an existing connection (if not it
+     * creates one), than sends the packet to the destination waiting for the
```
Comment (Polpetta) [inline src/endpoint/egress.h]:
```suggestion
     * creates one), then sends the packet to the destination waiting for the
```

File: src/endpoint/egress.h [hunk 20]
```diff
@@ -24,17 +26,49 @@
...
+     * response.
+     * @param pkt Packet to send
+     * @param pkt_len Lenght of the packet to send
+     * @param ce Information on the connection
+     * @param fd Socket used for the connection
+     */
     void manage_exiting_tcp_packets(unsigned char* pkt, size_t pkt_len,
                                     const ConnectionEntry& ce, socket_fd fd);
+    /**
+     * Method used to manage TCP connection with the destination.
```
Comment (Polpetta) [inline src/endpoint/egress.h:52]: Idem
```suggestion
     * The method handles a TCP connection with a defined destination
```

File: src/endpoint/egress.h [hunk 21]
```diff
@@ -24,17 +26,49 @@
...
+     * It create a connection and send data using raw sockets, than it waits for
```
Comment (Polpetta) [inline src/endpoint/egress.h]:
```suggestion
     * It creates a connection and it sends the data using raw sockets, then it waits for
```

File: src/endpoint/endpoint.cpp [hunk 22]
```diff
@@ -1,17 +1,13 @@
...
-//
-
 #include "endpoint.h"
 
 classifier::Classifier endpoint::Endpoint::classifier_;
 // TODO retrieve roulette address
-db::DBQuery* endpoint::Endpoint::roulette_ = new db::DBQuery("localhost", 57684);
-
+db::DBQuery *endpoint::Endpoint::roulette_ = new db::DBQuery("localhost",
+                                                             57684);
```
Comment (Polpetta) [inline src/endpoint/endpoint.cpp]: We should put this variables as part of the CMake configuration

File: src/endpoint/endpoint.cpp [hunk 23]
```diff
@@ -141,6 +138,6 @@ std::string endpoint::Endpoint::get_my_ip() const {
 
 // TODO to refactor
```
Comment (Polpetta) [inline src/endpoint/endpoint.cpp]: Did we finish the refactor?

File: src/endpoint/ingress.cpp [hunk 24]
```diff
@@ -190,85 +117,87 @@ void endpoint::Ingress::manage_entering_tcp_packets(void * mngmnt_args) {
...
     struct sfc_header header =
-            utils::sfc_header::SFCUtilities::retrieve_header((uint8_t*)args->pkt);
+            utils::sfc_header::SFCUtilities::retrieve_header(
+                    (uint8_t *) args->pkt);
 
     struct iphdr ip_hdr =
             utils::PacketUtils::retrieve_ip_header(
-                    (unsigned char*)(args->pkt + SFC_HDR));
+                    (unsigned char *) (args->pkt + SFC_HDR));
     // TODO think something more extensible to support more protocols
```
Comment (Polpetta) [inline src/endpoint/ingress.cpp:198]: Did we complete this TODO?

File: src/endpoint/ingress.h [hunk 25]
```diff
@@ -27,11 +24,43 @@ namespace endpoint {
 
 class Ingress : public Endpoint {
 private:
+    /**
+     * Method used to manage entering TCP connections.
```
Comment (Polpetta) [inline src/endpoint/ingress.h]:
```suggestion
     * Method that manages incoming TCP connections.
```

File: src/endpoint/ingress.h [hunk 26]
```diff
@@ -27,11 +24,43 @@ namespace endpoint {
...
+     * Once a packet is received it will be classified. After roulette will be
+     * updated to maintain data of the connection. To the packet is added the
+     * SFC header and it is sent to the chain.
+     * When the client close the connection it cleanup roulette db.
+     * @param args tcp_pkt_mngmnt_args struct that contains data to manage
+     * entering packets.
+     */
     void manage_entering_tcp_packets(void* args);
+    /**
+     * Method used to manage entering UDP connections.
```
Comment (Polpetta) [inline src/endpoint/ingress.h]:
```suggestion
     * Method that manages incoming UDP connections.
```

File: src/endpoint/ingress.h [hunk 27]
```diff
@@ -27,11 +24,43 @@ namespace endpoint {
...
+     * Once a packet is received it will be classified. After roulette will be
+     * updated to maintain data of the connection. To the packet is added the
+     * SFC header and it is sent to the chain.
+     * @param args udp_pkt_mngmnt_args struct that contains data to manage
+     * entering packets.
+     */
     void manage_entering_udp_packets(void* args);
+    /**
+     * Method used to manage UDP packets received from the chain.
```
Comment (Polpetta) [inline src/endpoint/ingress.h]:
```suggestion
     * Method that manages received UDP packets coming from the chain.
```

File: src/endpoint/utils/sfcfixedlengthheader.h [hunk 28]
```diff
@@ -4,24 +4,25 @@
...
 #include "sfcutilities.h"
 
 #include "config.h"
 
 namespace utils {
-
 namespace sfc_header {
 
+/**
+ * Class that represent the SFC header
```
Comment (Polpetta) [inline src/endpoint/utils/sfcfixedlengthheader.h]:
```suggestion
 * Class that represents the SFC header.
```

File: src/endpoint/utils/sfcutilities.h [hunk 29]
```diff
@@ -1,37 +1,75 @@
...
 
 #include "packetutils.h"
 #include "sfcheaderstruct.h"
+#include "log.h"
 
 namespace utils {
 namespace sfc_header {
 
+/**
+ * Utility class to manipulate SFC header
```
Comment (Polpetta) [inline src/endpoint/utils/sfcutilities.h]: This
```suggestion
 * Utility class to manipulate a SFC header
```
or
```suggestion
 * Utility class to manipulate SFC headers
```

File: src/server/server.cpp [hunk 30]
```diff
@@ -1,16 +1,15 @@
...
 
-server::Server::Server(uint16_t port) : port_(port){}
+server::Server::Server(uint16_t port) : port_(port) {}
 
 void server::Server::stop() {
     stopped_ = true;
 }
 
 void server::Server::signal_handler(int signal_number) {
     /* TODO: Put exit cleanup code here. */
```
Comment (Polpetta) [inline src/server/server.cpp:10]: Do we still need to clean up resources before exiting?

File: src/server/servertcp.cpp [hunk 31]
```diff
@@ -47,15 +43,17 @@ void server::tcp::ServerTCP::run() {
         /* TODO: malloc'ing before accepting a connection causes only one small
```
Comment (Polpetta) [inline src/server/servertcp.cpp:43]: Is this TODO resolvable?

File: src/server/utils/asynctaskexecutor.h [hunk 32]
```diff
@@ -27,8 +23,8 @@ class AsyncTaskExecutor {
     static AsyncTaskExecutor* inst;
 };
 
-#define ASYNC_TASK(task) \
-    utils::AsyncTaskExecutor::instance()->submit_task(task);
+/*#define ASYNC_TASK(task) \
+    utils::AsyncTaskExecutor::instance()->submit_task(task);*/
```
Comment (Polpetta) [inline src/server/utils/asynctaskexecutor.h]: Remove this macro if we don't need it anymore

File: src/server/utils/asynctaskexecutor.h [hunk 33]
```diff
@@ -27,8 +23,8 @@ class AsyncTaskExecutor {
...
 // TODO find something better
```
Comment (Polpetta) [inline src/server/utils/asynctaskexecutor.h:26]: Did you find something better?

**Review by Polpetta — COMMENTED** [review]
There are some little fixes to do, but generally speaking it's ok.

If you want, apply the suggestions I made. When you're done ping me and I'll merge this PR :+1:

File: src/classifier/classifier.cpp (hunk 1 above)
Comment (mzanella) [reply src/classifier/classifier.cpp:58]: Not yet. We can close it when we will implement call to harbor for the available chains.

File: src/classifier/classifier.cpp (hunk 2 above)
Comment (mzanella) [reply src/classifier/classifier.cpp:66]: Not yet. We can close it when we will implement call to harbor for the available chains.

File: src/db/dbquery.cpp (hunk 7 above)
Comment (mzanella) [reply src/db/dbquery.cpp]: done

File: src/db/dbquery.h (hunk 8 above)
Comment (mzanella) [reply src/db/dbquery.h]: done

File: src/db/dbquery.h (hunk 9 above)
Comment (mzanella) [reply src/db/dbquery.h]: done

File: src/db/dbquery.h (hunk 10 above)
Comment (mzanella) [reply src/db/dbquery.h]: done

File: src/db/dbquery.h (hunk 11 above)
Comment (mzanella) [reply src/db/dbquery.h]: done

File: src/db/dbquery.h (hunk 12 above)
Comment (mzanella) [reply src/db/dbquery.h]: done

File: src/db/utils/address.cpp (hunk 13 above)
Comment (mzanella) [reply src/db/utils/address.cpp]: done

File: src/db/dbquery.cpp (hunk 14 above)
Comment (mzanella) [reply src/db/dbquery.cpp]: done

File: src/endpoint/egress.cpp (hunk 15 above)
Comment (mzanella) [reply src/endpoint/egress.cpp]: done

File: src/endpoint/egress.cpp (hunk 16 above)
Comment (mzanella) [reply src/endpoint/egress.cpp:249]: More or less, connection cleanup is not treated correctly

File: src/endpoint/egress.h (hunk 17 above)
Comment (mzanella) [reply src/endpoint/egress.h]: done

File: src/endpoint/egress.h (hunk 18 above)
Comment (mzanella) [reply src/endpoint/egress.h]: done

File: src/endpoint/egress.h (hunk 19 above)
Comment (mzanella) [reply src/endpoint/egress.h]: done

File: src/endpoint/egress.h (hunk 20 above)
Comment (mzanella) [reply src/endpoint/egress.h:52]: done

File: src/endpoint/egress.h (hunk 21 above)
Comment (mzanella) [reply src/endpoint/egress.h]: done

File: src/endpoint/endpoint.cpp (hunk 22 above)
Comment (mzanella) [reply src/endpoint/endpoint.cpp]: Do you mean the pointer or localhost/57684?

File: src/endpoint/endpoint.cpp (hunk 23 above)
Comment (mzanella) [reply src/endpoint/endpoint.cpp]: Yes

File: src/endpoint/ingress.cpp (hunk 24 above)
Comment (mzanella) [reply src/endpoint/ingress.cpp:198]: Never, I hope we will only support TCP and UDP. This TODO it is for posterity.

File: src/endpoint/ingress.h (hunk 25 above)
Comment (mzanella) [reply src/endpoint/ingress.h]: done

File: src/endpoint/utils/sfcutilities.h (hunk 29 above)
Comment (mzanella) [reply src/endpoint/utils/sfcutilities.h]: oook

File: src/server/server.cpp (hunk 30 above)
Comment (mzanella) [reply src/server/server.cpp:10]: Probably

File: src/server/servertcp.cpp (hunk 31 above)
Comment (mzanella) [reply src/server/servertcp.cpp:43]: If you are brave enought

File: src/server/utils/asynctaskexecutor.h (hunk 33 above)
Comment (mzanella) [reply src/server/utils/asynctaskexecutor.h:26]: No way. I think that our work around is the only solution for our knowledge.

File: src/server/utils/asynctaskexecutor.h (hunk 32 above)
Comment (mzanella) [reply src/server/utils/asynctaskexecutor.h]: done

File: src/classifier/classifier.cpp (hunk 1 above)
Comment (Polpetta) [reply src/classifier/classifier.cpp:58]: ook

File: src/classifier/classifier.cpp (hunk 2 above)
Comment (Polpetta) [reply src/classifier/classifier.cpp:66]: ook

File: src/endpoint/egress.cpp (hunk 16 above)
Comment (Polpetta) [reply src/endpoint/egress.cpp:249]: Are you going to fix it this pr or do you think we can safely merge without this resolved for now?

File: src/endpoint/endpoint.cpp (hunk 22 above)
Comment (Polpetta) [reply src/endpoint/endpoint.cpp]: No, I mean something like
```C++
#define ROULETTE_DEFAULT_ADDRESS localhost
#define ROULETTE_DEFAULT_PORT 57687
```

In the `config.h.in` file

File: src/endpoint/ingress.cpp (hunk 24 above)
Comment (Polpetta) [reply src/endpoint/ingress.cpp:198]: Fair enough

File: src/server/server.cpp (hunk 30 above)
Comment (Polpetta) [reply src/server/server.cpp:10]: Are we going to fix this issue in this branch?

File: src/server/utils/asynctaskexecutor.h (hunk 32 above)
Comment (Polpetta) [reply src/server/utils/asynctaskexecutor.h]: Good boie

File: src/server/utils/asynctaskexecutor.h (hunk 33 above)
Comment (Polpetta) [reply src/server/utils/asynctaskexecutor.h:26]: Okok. Clean up the TODO then :man_shrugging:

File: src/endpoint/egress.cpp (hunk 16 above)
Comment (mzanella) [reply src/endpoint/egress.cpp:249]: I think it is better to manage all resource claenup in a new PR... This is only a code cleanup and warning correction PR

File: src/db/dbquery.cpp (hunk 3 above)
Comment (mzanella) [reply src/db/dbquery.cpp]: done

File: src/db/dbquery.cpp (hunk 4 above)
Comment (mzanella) [reply src/db/dbquery.cpp]: done

File: src/db/dbquery.cpp (hunk 5 above)
Comment (mzanella) [reply src/db/dbquery.cpp]: done

File: src/db/dbquery.cpp (hunk 6 above)
Comment (mzanella) [reply src/db/dbquery.cpp]: done

File: src/server/server.cpp (hunk 30 above)
Comment (mzanella) [reply src/server/server.cpp:10]: no

Comment (Polpetta) [reply src/server/server.cpp:10]: ok

File: src/endpoint/egress.cpp (hunk 16 above)
Comment (Polpetta) [reply src/endpoint/egress.cpp:249]: ok

**Pull request merged**